compiler/src/iree/compiler/Codegen/Common/GPU/GPUDistributionPatterns.cpp

```diff
@@ -643,6 +644,91 @@ struct DistributeTranspose final : OpDistributionPattern<vector::TransposeOp> {
   }
 };
 
+struct DistributeLayoutConflictResolutions final
+    : OpDistributionPattern<IREE::VectorExt::LayoutConflictResolutionOp> {
+  using OpDistributionPattern::OpDistributionPattern;
+
+  VectorValue reshapeVector(Location loc, RewriterBase &rewriter,
+                            VectorValue src, LayoutAttr &currentLayout,
+                            LayoutAttr &targetLayout, Type elementType) const {
+
+    SmallVector<int64_t> targetShape = targetLayout.getDistributedShape();
+    SmallVector<int64_t> currentShape = currentLayout.getDistributedShape();
+
+    auto newVectorType = VectorType::get(targetShape, elementType);
+    auto constantOp = rewriter.create<arith::ConstantOp>(
+        loc, newVectorType, rewriter.getZeroAttr(newVectorType));
+    auto newVector = dyn_cast<VectorValue>(constantOp.getResult());
+
+    int64_t innermostDim = targetShape.size() - 1;
+    int64_t step =
+        std::min(targetShape[innermostDim], currentShape[innermostDim]);
+    DenseMap<LayoutDimension, int64_t> steps;
+    LayoutDimension vecDim = LayoutDimension::VECTORX;
+    steps[vecDim] = step;
+    LayoutIterator srcIterator(currentLayout, steps);
+    LayoutIterator targetIterator(targetLayout, steps);
+
+    for (; !srcIterator.iterationComplete() &&
```

Comment:
nit: A while loop might be cleaner

Reply:
With the for loop, you get to keep the increments in one place too. I would prefer to keep things as is unless you feel strongly.

Reply:
For loop should be fine